### ISSUE TITLE
JSON format is not effective under certain circumstances

### ISSUE THREAD

**Issue by jwadolowski** [wata727/tflint#369]
# Steps to reproduce

Run `tflint -f json` for the following file:

```hcl
resource "aws_instance" "web" {
  ami           = "ami-12345678"
  instance_type = "t1.2xlarge"
}

resource "aws_iam_server_certificate" "test" {
  name_prefix       = "test"
  certificate_body  = "${file("../../not/existing/path/to/certificates/example.com/test.crt")}"
  certificate_chain = "${file("../../not/existing/path/to/certificates/example.com/intermediate.crt")}"
  private_key       = "${file("../../not/existing/path/to/certificates/example.com/test.key")}"

  lifecycle {
    create_before_destroy = true
  }
}
```

# Current state

```
$ tflint -f json test.tf
Error: Failed to check `aws_iam_server_certificate_invalid_certificate_body` rule: Failed to eval an expression in test.tf:8; Error in function call: Call to function "file" failed: no file exists at ../../not/existing/path/to/certificates/example.com/test.crt.
```

# Expected state

It'd be great to preserve user-defined output format regardless of detected error type.

# Noteworthy facts

If we take problematic resource out of the file and run `tflint -f json` again it works just fine.

```hcl
resource "aws_instance" "web" {
  ami           = "ami-12345678"
  instance_type = "t1.2xlarge"
}
```

Here's the output `tflint -f json` generates:

```json
[
  {
    "detector": "aws_instance_invalid_type",
    "type": "ERROR",
    "message": "instance_type is not a valid value",
    "line": 3,
    "file": "test.tf",
    "link": ""
  }
]
```

I use `tflint` as a [vim linter](https://github.com/w0rp/ale/blob/master/ale_linters/terraform/tflint.vim), which expects JSON to work properly.

**Comment by wata727** [comment]
That makes sense. I'm planning to change the output format in https://github.com/wata727/tflint/issues/247. This proposal will be included in the change.

Btw, I'm working on [LSP support](https://github.com/wata727/tflint/issues/275) now. If you want to improve the experience in your editor, this may be more appropriate.
